lib/logpipe.h

```diff
@@ -363,6 +366,16 @@ log_pipe_append(LogPipe *s, LogPipe *next)
   s->pipe_next = next;
 }
 
+void
+log_pipe_set_persist_id(LogPipe * const self, const gchar * const persist_id);
+
+static inline void
+log_pipe_generate_persist_id(LogPipe * const self)
+{
+  if (self->persist_id != NULL && self->generate_persist_id != NULL)
+    self->persist_id = self->generate_persist_id(self);
```

Comment:
`generate_persist_id()` returns a different kind of `const` than the `persist_id` field